animations/lib/src/basics/05_animated_builder.dart

```diff
@@ -16,8 +16,8 @@ class _AnimatedBuilderDemoState extends State<AnimatedBuilderDemo>
   static const Color beginColor = Colors.deepPurple;
   static const Color endColor = Colors.deepOrange;
   Duration duration = Duration(milliseconds: 800);
-  AnimationController controller;
-  Animation<Color> animation;
+  late AnimationController controller;
+  late Animation<Color?> animation;
```

Comment:
I'm curious how nulls could be inserted this animation.

Reply:
While doing the conversion, this was a weird part for me as well. `ColorTween` is a glorified version of `Tween<Color?>`, which does allow for a null `Color` value:

https://github.com/flutter/flutter/blob/7c0300a3588d91542366d9efadda26fa2c9d9224/packages/flutter/lib/src/animation/tween.dart#L355

I thought about switching to `Tween<Color>`, but that would cause us to lose the `Color`-specific `lerp` function provided by `ColorTween`.

In the case of this example, I don't expect the tween to ever produce a null value, but it looks like `MaterialButton` is perfectly happy to accept one and just use the default color in response:

https://github.com/flutter/flutter/blob/88809aa24720705d1cd8de0bfc46faf932fe0b6e/packages/flutter/lib/src/material/material_button.dart#L180